### PR TITLE
customcluster-controller-UT

### PR DESCRIPTION
**Pull request by LiZhenCheng9527**
some UT for customCluster_controller.go

### CODE REVIEW

File: pkg/cluster-operator/customcluster_controller_test.go
Comment (hzxuzhonghu) [inline pkg/cluster-operator/customcluster_controller_test.go:1]: add copy right

File: pkg/cluster-operator/customcluster_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fake "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestDeleteWorkerPods(t *testing.T) {
+	//create a fake client
+	r := &CustomClusterController{}
+	r.Client = fake.NewClientBuilder().Build()
+	TestcustomCluster := &v1alpha1.CustomCluster{}
```
Comment (hzxuzhonghu) [inline pkg/cluster-operator/customcluster_controller_test.go:21]: testCustomCluster

File: pkg/cluster-operator/customcluster_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+	err := r.deleteWorkerPods(context.Background(), TestcustomCluster)
```
Comment (hzxuzhonghu) [inline pkg/cluster-operator/customcluster_controller_test.go:23]: I am not sure what this test do as `TestcustomCluster` actually has no name

File: pkg/cluster-operator/customcluster_controller_test.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+			},
+		},
+	}
+
+	r := &CustomClusterController{}
+	//r.Client = fake.NewClientBuilder().WithObjects(Cluster).Build()
+
+	requests := r.ClusterToCustomClusterMapFunc(Cluster)
+	//expectRequests := []ctrl.Request{{NamespacedName: client.ObjectKey{Namespace: "test", Name: "test"}}}
+	assert.Equal(t, 1, len(requests))
```
Comment (hzxuzhonghu) [inline pkg/cluster-operator/customcluster_controller_test.go:99]: Why donot check the result here

File: pkg/cluster-operator/customcluster_controller_test.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"kurator.dev/kurator/pkg/apis/infra/v1alpha1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fake "sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestDeleteWorkerPods(t *testing.T) {
+	//create a fake client
```
Comment (Xieql) [inline pkg/cluster-operator/customcluster_controller_test.go:18]: nit：add a space after the double slashes (//) for better readability
replace `//create a fake client` with `// create a fake client`
and check the others

File: pkg/cluster-operator/customcluster_controller_test.go [hunk 5]
```diff
@@ -0,0 +1,100 @@
+package clusteroperator
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"kurator.dev/kurator/pkg/apis/infra/v1alpha1"
```
Comment (Xieql) [inline pkg/cluster-operator/customcluster_controller_test.go:10]: the package 'kurator.dev/kurator/pkg/apis/infra/v1alpha1' needs to be separated from the others.
see https://github.com/kurator-dev/kurator/actions/runs/5797336703/job/15732862721#step:6:177

File: pkg/cluster-operator/customcluster_controller_test.go [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+			InfrastructureRef: &corev1.ObjectReference{
+				Kind:      CustomClusterKind,
+				Namespace: "test",
+				Name:      "test-cluster",
+			},
+		},
+	}
+
+	r := &CustomClusterController{}
+	//r.Client = fake.NewClientBuilder().WithObjects(Cluster).Build()
```
Comment (Xieql) [inline pkg/cluster-operator/customcluster_controller_test.go:95]: The code should be removed instead of being commented out if it is not needed anymore.

**Pull request closed without merging**